lesson22/src/js/main.js

```diff
@@ -0,0 +1,241 @@
+        const usersData = json.data;
+
+        if (!usersData.length) {
+            renderErrorMessage("No user.",table);
+            return;
+        }
+        renderTableElements(usersData);
+    }catch(error){
+        console.error(error);
+    }finally{
+       loading.removeLoading(table);
+    }
+}
+
+const renderTableElements = usersData => {
+    renderTableHeader(usersData);
+    renderTableBody(usersData);
+}
+
+const tableColumnConfig = {
+    id: {
+      value: "ID",
+      hasSort: true
+    },
+    name: {
+      value: "名前",
+      hasSort: false
+    },
+    gender: {
+      value: "性別",
+      hasSort: false
+    },
+    age: {
+      value: "年齢",
+      hasSort: true
+    }
+  };
+
+const renderTableHeader = (usersData) => {
+    const thead = document.createElement("thead");
+    const tr = document.createElement("tr");
+    thead.className = "bg-slate-500";
+
+    for (const [columnKey, columnValue] of Object.entries(tableColumnConfig)) {
+        const th = createAttributedElements({
+            tag:"th",
+            valuesByAttributes:{
+                class:"text-sm text-white px-6 py-4",
+            },
+            str:columnValue.value
+        });
+
+        tr.appendChild(th);
+
+        if (columnValue.hasSort) {
+            th.appendChild(createSortButtons(columnKey,usersData));
+        }
+    };
+
+    table.appendChild(thead).appendChild(tr);
+}
+
+const renderTableBody = usersData => {
+    const tbody = document.createElement("tbody");
+    tbody.id = "js-usersTableBody";
+
+    for(const user of usersData){
+        const fragment = document.createDocumentFragment();
+        const tr = document.createElement("tr");
+
+        Object.keys(tableColumnConfig).forEach((column) => {
+            const td = createAttributedElements({
+                tag:"td",
+                valuesByAttributes:{
+                    class:"border border-gray-300 px-4 py-2"
+                },
+                str:user[column]
+            })
+            fragment.appendChild(tr).appendChild(td);
+        });
+
+        table.appendChild(tbody).appendChild(fragment);
+    }
+}
+
+const sortButtonAttributes = [
+    {state:"asc",src:"../img/asc.svg", alt:"asc-image"},
+    {state:"desc",src:"../img/desc.svg",alt:"desc-image"},
+    {state:"both",src:"../img/both.svg",alt:"both-image"}
+]
+
+const createSortButtons = (columnKey,usersData) => {
+    const sortButtonsBox = createAttributedElements ({
+        tag:"div",
+        valuesByAttributes:{
+            id:`js-${columnKey}Buttons-Box`,
+            class: "w-10 inline-block align-middle js-sortButtons-Box"
+        }
+    })
+
+    const fragment = document.createDocumentFragment();
+
+    for(const button of sortButtonAttributes){
+        const sortButton = createAttributedElements({
+            tag:"button",
+            valuesByAttributes:{
+                class: `js-sortButtons-Item`,
```

Comment:
nit
```suggestion
                class: `js-sortButtons-item`,
```

Reply:
Thanks for letting me know! 
I fixed.
https://github.com/nakagawamai/js-lesson/pull/24/commits/67e4b4ae77ab50bf8ae81172a05128bee443e717